lib/client/kubesession.go

```diff
@@ -60,10 +61,20 @@ func NewKubeSession(ctx context.Context, tc *TeleportClient, meta types.SessionT
 	ws, resp, err := dialer.Dial(joinEndpoint, nil)
 	defer resp.Body.Close()
 	if err != nil {
-		body, _ := io.ReadAll(resp.Body)
-		fmt.Printf("Handshake failed with status %d\nand body: %v\n", resp.StatusCode, string(body))
 		cancel()
-		return nil, trace.Wrap(err)
+		body, _ := io.ReadAll(resp.Body)
+		var respData map[string]interface{}
+		if err := json.Unmarshal(body, &respData); err != nil {
+			return nil, trace.Wrap(err)
+		}
+
+		if message, ok := respData["message"]; ok {
+			if message, ok := message.(string); ok {
+				return nil, trace.Errorf("%v", message)
+			}
+		}
+
+		return nil, trace.BadParameter("failed to decode remote error: %v", string(body))
 	}
```

Comment:
Nit: looks to me like we now always omit the error returned by `Dial`.  Maybe if the body is empty we should be returning that error?  I assume if we didn't get far enough to have a proper request body, that error might be the more meaningful one, yeah?